pkg/resourceinterpreter/configurableinterpreter/interpreters.go

```diff
@@ -0,0 +1,212 @@
+package configurableinterpreter
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/resourceinterpreter/configurableinterpreter/engine"
+)
+
+type Interpreter interface {
+	IsEnabled(operation configv1alpha1.InterpreterOperation) bool
+
+	Retain(desired *unstructured.Unstructured, observed *unstructured.Unstructured) (retained *unstructured.Unstructured, enabled bool, err error)
+
+	GetReplicas(object *unstructured.Unstructured) (replicas int32, require *workv1alpha2.ReplicaRequirements, enabled bool, err error)
+
+	ReviseReplica(object *unstructured.Unstructured, replica int64) (obj *unstructured.Unstructured, enabled bool, err error)
+
+	AggregateStatus(object *unstructured.Unstructured, aggregatedStatusItems []workv1alpha2.AggregatedStatusItem) (obj *unstructured.Unstructured, enabled bool, err error)
+
+	GetDependencies(object *unstructured.Unstructured) (dependencies []configv1alpha1.DependentObjectReference, enabled bool, err error)
+
+	ReflectStatus(object *unstructured.Unstructured) (status *runtime.RawExtension, enabled bool, err error)
+
+	InterpretHealth(object *unstructured.Unstructured) (healthy bool, enabled bool, err error)
+}
+
+type InterpreterManager interface {
```

Comment:
Could you please add a comment for the `InterpreterManager`? Including what it is exactly and what's the responsibilities.

I suppose it is an `executor` for a specified kind of resource. It provides `LoadConfig` interface to accept the rules, and expose methods to do the interpreter work.

Reply:
Updated